### PR TITLE
Card images

### PR DESCRIPTION
**Pull request by britt6612**
## Work Done 

I added preview components to 
- [x] Anchor
- [ ] Accordion
- [x] Box
- [x] Button
- [x] CheckBox
- [x] Footer
- [x] Grid
- [x] Header
- [x] Layer
- [ ] Main
- [x] MakedInput
- [x] Menu
- [x] RadioButton Group
- [x] RangeInput
- [x] Select
- [x] Stack
- [x] Tabs
- [x] TextArea
- [x] TextInput

## To do 

* I need to find previews that are easy to use for the remaining components 
* Also having trouble fitting Accordion into the box so looking at options for that 
* Issue #658 
* Also will handle accordion in other PR Tile Preview since we are using svg's for that image

### CODE REVIEW

File: aries-site/src/data/structure.js [hunk 1]
```diff
@@ -323,6 +340,11 @@ export const structure = [
     seoDescription:
       'Used with text based navigation, such as inline text, header navigation, and footer navigation.',
     sections: [],
+    previewComponent: {
+      component() {
```
Comment (taysea) [inline aries-site/src/data/structure.js:344]: any reason this one doesn't follow same `() => <AmchorPreview />` structure as others?

File: aries-site/src/data/structure.js [hunk 2]
```diff
@@ -347,6 +377,10 @@ export const structure = [
     seoDescription:
       'When the user needs to select one or more options, use a checkbox.',
     sections: [],
+    previewComponent: {
+      component: () => <CheckBoxPreview />,
+      cardColor: 'background-front',
```
Comment (taysea) [inline aries-site/src/data/structure.js:382]: It feels like most cards want the background to match the color of the content card. What if by default the previews used `background-front` (instead of `background-contrast`), then in the places we want to modify it and need background-contrast we can add it in with the cardColor. It mights save us some lines of code.

File: aries-site/src/data/structure.js [hunk 3]
```diff
@@ -399,12 +441,19 @@ export const structure = [
...
+    },
   },
   {
     name: 'Menu',
     description: 'Menu is used to filter or sort content on a page.',
     seoDescription: 'Menu is used to filter or sort content on a page.',
     sections: [],
+    previewComponent: {
+      component: () => <MenuPreview />,
```
Comment (taysea) [inline aries-site/src/data/structure.js:455]: Not sure if this will be helpful or not, but what if ContentCard checked for the `typeof` previewComponent. If it's an object, then it functions as you currently have it set up. But, if it's just a function it just renders the component. That way, for previewComponents with no extra styling, you could just have `previewComponent: () => <MenuPreview />`

Soft opinion here.

Comment (taysea) [reply aries-site/src/data/structure.js:455]: But if we lean towards having the default background be background front, then it might save some code.

File: aries-site/src/examples/cardPreviews/header.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { Box, Header, Button, Text } from 'grommet';
+import { Hpe, Menu } from 'grommet-icons';
+
+export const HeaderPreview = () => {
+  return (
+    <Header background="background-front" fill="horizontal">
```
Comment (taysea) [inline aries-site/src/examples/cardPreviews/header.js:7]: Can we add a bottom border here to match the designs a little closer? ignore the blue strip on the side, that was just in the screen shot
Screen Shot 2020-04-07 at 6 17 59 PM

File: aries-site/src/tests/accessibility/axe.js [hunk 5]
```diff
@@ -24,7 +24,23 @@ test('should check Home page', async t => {
   // Only need to run in one browser
   if (t.browser.name === 'Chrome') {
     const axeContext = {
-      exclude: [],
+      exclude: [
```
Comment (taysea) [inline aries-site/src/tests/accessibility/axe.js:27]: Are these the label exclusions?

File: aries-site/src/examples/cardPreviews/header.js (hunk 4 above)
Comment (britt6612) [reply aries-site/src/examples/cardPreviews/header.js:7]: your eyes are better then Mine yes I will add that I wonder if there was also one on the footer I can add to make it feel separate

File: aries-site/src/data/structure.js (hunk 1 above)
Comment (britt6612) [reply aries-site/src/data/structure.js]: nope lol just had that at the beginning and forgot to change back

File: aries-site/src/tests/accessibility/axe.js (hunk 5 above)
Comment (britt6612) [reply aries-site/src/tests/accessibility/axe.js:27]: yes these are the labels that were in the front tiles that I had to exclude I wasnt sure if this was a good approach for now

File: aries-site/src/components/cards/ContentCard.js [hunk 6]
```diff
@@ -21,8 +21,29 @@ export const ContentCard = forwardRef(({ topic, ...rest }, ref) => {
       {...rest}
     >
       <Box gap="large">
+        {previewComponent && (
+          <Box
+            background={previewComponent.cardColor || 'background-contrast'}
```
Comment (halocline) [inline aries-site/src/components/cards/ContentCard.js:26]: What does `cardColor` represent? To me, it represents the background color for the preview Box, correct? If so, could we use `background` instead of `cardColor` to be more precise on what this property holds?

Comment (halocline) [inline aries-site/src/components/cards/ContentCard.js:26]: Can we add a
```
ContentCard.defaultProps = {
  previewComponent: {
    cardColor: 'background-contrast'
  }
}
``` 
Then remove the OR condition here?

File: aries-site/src/components/cards/ContentCard.js [hunk 7]
```diff
@@ -21,8 +21,29 @@ export const ContentCard = forwardRef(({ topic, ...rest }, ref) => {
...
+            height="small"
+            round="xsmall"
+            overflow="hidden"
+            style={{ position: 'relative' }}
+          >
```
Comment (halocline) [inline aries-site/src/components/cards/ContentCard.js:31]: It seems like this is duplicative of lines 45-51 and should be combined. Then the conditional should move inside this box and present either the previewComponent or the previewImage.

File: aries-site/src/components/cards/ContentCard.js [hunk 8]
```diff
@@ -61,6 +83,11 @@ ContentCard.propTypes = {
       icon: PropTypes.func.isRequired,
       name: PropTypes.string.isRequired,
     }),
+    previewComponent: PropTypes.shape({
+      component: PropTypes.func,
+      cardColor: PropTypes.string,
+      justify: PropTypes.string,
+    }),
```
Comment (halocline) [inline aries-site/src/components/cards/ContentCard.js:90]: Related to @taysea's https://github.com/hpe-design/design-system/pull/647#discussion_r405203370, I wonder if this becomes
```suggestion
    previewComponent: PropTypes.oneOfType([
      PropTypes.elementType,
      PropTypes.shape({
        component: PropTypes.func,
        cardColor: PropTypes.string,
        justify: PropTypes.string,
      })
    ]),
```

File: aries-site/src/data/structure.js [hunk 9]
```diff
@@ -7,6 +7,23 @@ import {
...
+  HeaderPreview,
+  MaskedInputPreview,
+  MenuPreview,
+  RadioButtonGroupPreview,
+  RangeInputPreview,
+  SelectPreview,
+  TabsPreview,
+  TextAreaPreview,
+  TextInputPreview,
+} from '../examples/cardPreviews';
```
Comment (halocline) [inline aries-site/src/data/structure.js:26]: Question for all @britt6612 @taysea related to project structure... is it more helpful to think of the "preview" belonging to the component example? Or to cardPreviews? If it belonged to the example, then I can easily see if that example has a preview or is missing it. Otherwise, I have to go and look for another directory which I may or may not know exists (assuming I am unfamiliar with the codebase).

In more concrete terms, would we rather have a structure like this:
/src
|--examples
|   |--components
|   |   |--button
|   |   |   |--ButtonExample.js
|   |   |   |--ButtonIconExample.js
|   |   |   |--ButtonPreview.js
|   |   |   |--index.js
|   |   |--checkbox
|   |   |   |--CheckboxExample.js
|   |   |   |--CheckboxPreview.js
|   |   |   |--index.js
|   |   ...
|   |--foundation
|   |--templates

Or like how it is now?
/src
|--examples
|   |--cardPreviews
|   |   |--ButtonPreview.js
|   |   |--CheckboxPreview.js
|   |   |--index.js
|   |   |  ...
|   |--components
|   |   |--button
|   |   |   |--ButtonExample.js
|   |   |   |--ButtonIconExample.js
|   |   |   |--index.js
|   |   |--checkbox
|   |   |   |--CheckboxExample.js
|   |   |   |--index.js
|   |   ...
|   |--foundation
|   |--templates

If we adopted the former, that work could be handled outside of this PR.

File: aries-site/src/data/structure.js [hunk 10]
```diff
@@ -355,14 +387,18 @@ export const structure = [
...
     name: 'Grid',
     description:
       'The Grid component is used to layout content. Responsive grid is important to consider in every use case.',
-    previewImage: {
-      src: '/grid-light.svg',
-      alt: 'HPE Grid Preview',
+    previewComponent: {
+      component: () => <GridPreview />,
+      cardColor: 'background-front',
     },
```
Comment (halocline) [inline aries-site/src/data/structure.js:402]: Wondering if we can collapse the `previewImage` and `previewComponent` into a single `preview` property? These objects could then look like this:
```
{
  name: 'Grid',
  description: 'dfasdfadsf fsdafdsadff fdsafda',
  ...
  preview: {
    component: () => <GridPreview />,
    background: 'background-front',
    image: {
       src: '/grid-light.svg',
       alt: 'HPE Grid Preview',
    },
    justify: 'efadfnd'
  },
},
```

File: aries-site/src/data/structure.js (hunk 9 above)
Comment (britt6612) [reply aries-site/src/data/structure.js:26]: Initially I did think about putting the Previews within the examples as you have above in the first example. Some of them I am using the examples since if they are simple enough so having them in the same section may be easier for a new user to realize here is a preview and here is the example then also by that know that if there is no preview then the example was most likely used

Comment (taysea) [reply aries-site/src/data/structure.js:27]: I kind of like the first option as well, easier for scan-ability. I agree, I'd be fine updating structure in a separate PR.

File: aries-site/src/data/structure.js (hunk 10 above)
Comment (taysea) [reply aries-site/src/data/structure.js]: I like this idea -- I would assume a preview would have either a component or an image (not both) -- and we could check for that in ContentCard.

File: aries-site/src/components/cards/ContentCard.js [hunk 11]
```diff
@@ -21,21 +22,28 @@ export const ContentCard = forwardRef(({ topic, ...rest }, ref) => {
...
+                  flex
+                  justify={preview.justify || 'center'}
+                  align="center"
+                >
+                  {preview.component()}
+                </Box>
+              ))}
+          </PreviewImageCard>
+        )}
+        {!preview && <PreviewImageCard />}
```
Comment (taysea) [inline aries-site/src/components/cards/ContentCard.js:46]: Do we have to have this check? Wouldn't the condition on 27 just have undefined if there was no preview?

File: aries-site/src/components/cards/PreviewCard.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+      style={{ position: 'relative' }}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+};
+
+PreviewImageCard.defaultProps = {
+  background: 'background-contrast',
```
Comment (taysea) [inline aries-site/src/components/cards/PreviewCard.js:20]: Does this differ from the background colors of the svgs? If so, maybe we should be those match.

File: aries-site/src/examples/cardPreviews/layer.js [hunk 13]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { Box } from 'grommet';
+
+export const LayerPreview = () => {
+  return (
+     <Box fill direction='row'>
+         <Box basis='3/4' background="brand"/>
```
Comment (taysea) [inline aries-site/src/examples/cardPreviews/layer.js:7]: This feels a bit loud in comparison to the rest. What if we used `background-contrast` or something instead?

File: aries-site/src/examples/cardPreviews/textinput.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { TextInput, FormField } from 'grommet';
+
+export const TextInputPreview = () => {
+  return (
+    <FormField
+      style={{ borderRadius: '4px' }}
+      fill="horizontal"
+      border={{ color: 'focus' }}
```
Comment (taysea) [inline aries-site/src/examples/cardPreviews/textinput.js:9]: Any way we can make this border a little thicker? it feels more narrow than a typical focus outline

File: aries-site/src/tests/accessibility/axe.js (hunk 5 above)
Comment (taysea) [reply aries-site/src/tests/accessibility/axe.js:27]: Great, maybe @halocline can give some advice of how to achieve labels in the DOM but visually hidden? would like to avoid excluding things from the tests wherever possible.

File: aries-site/src/data/structure.js (hunk 10 above)
Comment (taysea) [reply aries-site/src/data/structure.js]: ^ My comment can be ignored, looks like Brittany already implemented it nicely : )

File: aries-site/src/tests/accessibility/axe.js (hunk 5 above)
Comment (britt6612) [reply aries-site/src/tests/accessibility/axe.js:27]: yes! Totally forgot to discuss that with @halocline we can talk about that tomorrow

File: aries-site/src/components/cards/ContentCard.js (hunk 11 above)
Comment (britt6612) [reply aries-site/src/components/cards/ContentCard.js:46]: unless Im missing something yeah otherwise it would put another card underneath even the ones with the preview ? unless you meant another way

File: aries-site/src/data/structure.js (hunk 9 above)
Comment (britt6612) [reply aries-site/src/data/structure.js:27]: Ill create an issue for re structuring these files!

**Comment by britt6612** [comment]
@taysea thanks for fixing that now I see what you were referring to lol

**Comment by taysea** [comment]
> @taysea thanks for fixing that now I see what you were referring to lol

Aside from the labels thing, I'm good to go. I'm fine to merge as is and sort out the exclusions later.

**Comment by britt6612** [comment]
> > @taysea thanks for fixing that now I see what you were referring to lol
> 
> Aside from the labels thing, I'm good to go. I'm fine to merge as is and sort out the exclusions later.

Thank you I added to the above 
## To Do 
* Issue #658 

which will take care of that

File: aries-site/src/tests/accessibility/axe.js (hunk 5 above)
Comment (britt6612) [reply aries-site/src/tests/accessibility/axe.js]: It was the adding the `aria-label` instead of excluding these from the tests I just added these labels where they needed to be! @taysea @halocline

File: aries-site/src/examples/cardPreviews/footer.js [hunk 15]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Box, Footer, Text } from 'grommet';
+
+export const FooterPreview = () => {
+  return (
+    <Footer background="background-front" fill="horizontal">
```
Comment (halocline) [inline aries-site/src/examples/cardPreviews/footer.js:6]:
```suggestion
    <Footer background="background-front" border={{side: 'top' }} fill="horizontal">
```

File: aries-site/src/examples/cardPreviews/header.js [hunk 16]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Box, Header, Button, Text } from 'grommet';
+import { Hpe, Menu } from 'grommet-icons';
+
+export const HeaderPreview = () => {
+  return (
+    <Header
+      border={{ side: 'bottom' }}
```
Comment (halocline) [inline aries-site/src/examples/cardPreviews/header.js:8]:
```suggestion
      border={{ color: 'background-contrast', side: 'bottom' }}
```

File: aries-site/src/examples/cardPreviews/anchor.js [hunk 17]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Anchor, Box, Text } from 'grommet';
+
+export const AnchorPreview = () => {
+  return (
+    <Box direction="row" gap="xsmall">
+      <Text>The ship's</Text>
+      {/* Placing as span to avoid nested Anchor errors in DOM. Since this is
+       purely a visual preview, functionality isn't important here */}
+      <Anchor as="span" label="Anchor." />
```
Comment (halocline) [inline aries-site/src/examples/cardPreviews/anchor.js:10]:
```suggestion
      <Anchor as="span" label="Anchor" />.
```

File: aries-site/src/data/structure.js [hunk 18]
```diff
@@ -323,6 +362,9 @@ export const structure = [
     seoDescription:
       'Used with text based navigation, such as inline text, header navigation, and footer navigation.',
     sections: [],
+    preview: {
+      component: () => <AnchorPreview />,
+    },
```
Comment (halocline) [inline aries-site/src/data/structure.js:367]:
```suggestion
    preview: {
      component: () => <AnchorPreview />,
      background: 'background-front',
    },
```

File: aries-site/src/data/structure.js [hunk 19]
```diff
@@ -307,10 +344,12 @@ export const structure = [
   {
     name: 'Accordion',
     description: 'Buttons are used to indicate actions that can be performed.',
-    previewImage: {
-      src: '/accordion-light.svg',
-      alt: 'HPE Accordion Preview',
-      fit: 'contain',
+    preview: {
+      image: {
```
Comment (halocline) [inline aries-site/src/data/structure.js:348]: Looks like we are still using the image here rather than the preview component

File: aries-site/src/data/structure.js [hunk 20]
```diff
@@ -399,12 +467,19 @@ export const structure = [
...
+    },
   },
   {
     name: 'Menu',
     description: 'Menu is used to filter or sort content on a page.',
     seoDescription: 'Menu is used to filter or sort content on a page.',
     sections: [],
+    preview: {
+      component: () => <MenuPreview />,
+    },
```
Comment (halocline) [inline aries-site/src/data/structure.js:482]:
```suggestion
    preview: {
      component: () => <MenuPreview />,
      background: 'background-front',
    },
```

File: aries-site/src/data/structure.js [hunk 21]
```diff
@@ -413,6 +488,9 @@ export const structure = [
     seoDescription:
       'When one option of a set of options can be specified, use the RadioButtonGroup component.',
     sections: [],
+    preview: {
+      component: () => <RadioButtonGroupPreview />,
+    },
```
Comment (halocline) [inline aries-site/src/data/structure.js:493]:
```suggestion
    preview: {
      component: () => <RadioButtonGroupPreview />,
      background: 'background-front',
    },
```

File: aries-site/src/data/structure.js [hunk 22]
```diff
@@ -429,11 +511,17 @@ export const structure = [
     seoDescription:
       'The Select component is flexible to provide multiple select, search, and create options.',
     sections: [],
+    preview: {
+      component: () => <SelectPreview />,
+    },
```
Comment (halocline) [inline aries-site/src/data/structure.js:516]:
```suggestion
    preview: {
      component: () => <SelectPreview />,
      background: 'background-front',
    },
```

File: aries-site/src/data/structure.js [hunk 23]
```diff
@@ -429,11 +511,17 @@ export const structure = [
...
   },
   {
     name: 'Stack',
     description:
       'A Stack component is a container that stacks content on top of each other.',
+      preview: {
+        component: () => <StackExample />,
+      },
```
Comment (halocline) [inline aries-site/src/data/structure.js:524]:
```suggestion
      preview: {
        component: () => <StackExample />,
        background: 'background-front',
      },
```

File: aries-site/src/data/structure.js (hunk 19 above)
Comment (britt6612) [reply aries-site/src/data/structure.js:348]: Since accordion was supplied to me in the svgs I was going to change this in the other PR since I have now both light and dark versions but either way we can change it here now as well

Comment (halocline) [reply aries-site/src/data/structure.js:348]: Is there an issue representing this work? As long as its documented I am okay, but could get lost if someone else were to pick it up. There is a preview example file for it here: https://github.com/hpe-design/design-system/blob/7a8a3ab077002c817f0c6e6a479adf8d595821ee/aries-site/src/examples/cardPreviews/accordion.js
So it's a little hard to follow along from a review standpoint.

File: aries-site/src/components/cards/PreviewCard.js [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+      height="small"
+      round="xsmall"
+      overflow="hidden"
+      style={{ position: 'relative' }}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+};
```
Comment (halocline) [inline aries-site/src/components/cards/PreviewCard.js:17]: Makes the background prop a bit more explicit that it is a primary prop to be manipulated.
```suggestion
export const PreviewImageCard = ({ background, children, ...rest }) => {
  return (
    <Box
      background={background}
      height="small"
      round="xsmall"
      overflow="hidden"
      style={{ position: 'relative' }}
      {...rest}
    >
      {children}
    </Box>
  );
};
```

File: aries-site/src/components/cards/PreviewCard.js [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+
+PreviewImageCard.defaultProps = {
+  background: 'background-back',
+};
+
+PreviewImageCard.propTypes = {
+  children: PropTypes.node,
+  background: PropTypes.string,
```
Comment (halocline) [inline aries-site/src/components/cards/PreviewCard.js:25]: Do we want to accept { dark: 'valeu', light: 'value' } as a type for background?  If so,
```suggestion
  background: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.shape({
          dark: PropTypes.string,
          light: PropTypes.string,
      }),
  ]),
```

File: aries-site/src/data/structure.js (hunk 19 above)
Comment (britt6612) [reply aries-site/src/data/structure.js:348]: I can document it! Im so glad you pointed out that file I can go ahead and delete that one

File: aries-site/src/examples/cardPreviews/anchor.js (hunk 17 above)
Comment (britt6612) [reply aries-site/src/examples/cardPreviews/anchor.js:10]: opps lol

File: aries-site/src/components/cards/PreviewCard.js (hunk 25 above)
Comment (britt6612) [reply aries-site/src/components/cards/PreviewCard.js:26]: I think since for background we are using either `background-front` or `background-back` which already have their own light and dark in the theme it is fine. to just leave it as. one string being passed through. That was my though process.

**Comment by britt6612** [comment]
> A couple of clean up items, plus suggesting consistency among backgrounds as much as possible. I was seeing some inputs presented on white backgrounds, sometimes on grey and I could not see the rationale. I believe the designs are leaning toward the white/'background-front', but perhaps there was a discussion I missed.

I think you are correct the way I saw was the inputs had the white backgrounds. I was more or less guessing on some of the components that were not in the designs and was going to wait on feedback about those from Chris. I believe the pattern was to have the white background for the inputs it does not look appealing with the darker backgrounds. I will send him new link and get that checked off by him.

File: aries-site/src/data/structure.js (hunk 19 above)
Comment (britt6612) [reply aries-site/src/data/structure.js:348]: Added to issue #589

File: aries-site/src/components/cards/PreviewCard.js (hunk 25 above)
Comment (halocline) [reply aries-site/src/components/cards/PreviewCard.js:26]: I'm okay with this for now. We can modify in the future if it is too restrictive.

**Comment by britt6612** [comment]
Thank you @taysea & @halocline merging!
